File: python/leetcode3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        left=0
        set1=set() # set to store substring. 
        maxlength=0
        for right in range(len(s)):
            while s[right] in set1:
                set1.remove(s[left])
                left+=1

            set1.add(s[right])
            maxlength=max(maxlength,right-left+1)

        
        return maxlength

File: java/leetcode3.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        
        int left=0;
        int maxlength=0;
        Set<Character> charset = new HashSet<>();
        for(int right=0;right<s.length();right++){
            while(charset.contains(s.charAt(right))){
                charset.remove(s.charAt(left));
                left++;
            }
            charset.add(s.charAt(right));
            maxlength=Math.max(maxlength,right-left+1);
        }

        return maxlength;
    }
}